### ISSUE TITLE
Different layouts for the explore mode breadcrumbs

### ISSUE THREAD

**Issue by christianp** [numbas/Numbas#902]
At the moment, the navigation for explore mode parts shown at the top of the question is a tree, laid out horizontally. When there are lots of parts, it doesn't look great.

There should be a few different layout options, for different uses of explore mode.

## Uses of explore mode

* Linear - a sequence of parts, each building on the last, with no branching
* Small branches - A sequence, with occasional branches that are one or two parts deep, for things like hints
* Spoke and hub - Many independent branches off a central part.
* Long branches - Several branches with many steps, for different ways of doing the same activity.
* Loops - A state machine, returning to the same part(s) several times, for things like carrying out an iterative or recursive algorithm.

## Layouts

* Horizontal tree - the current layout
* Vertical tree
* Floaty graph - each part is a vertex, with directed edges showing the path
* Local view - Just the current part, the previous part, and next parts

**Comment by christianp** [comment]
It would be nice, for completely linear questions, to show all the parts at once.

**Comment by aharjula** [comment]
It will be really interesting to see how that would work (the technical side is probably easy for Numbas but I wonder about the cognitive load side for the student). In my world, a typical linear question has values, or things constructed from them, present from the previous parts, and if those latter parts are visible and those values are being updated "live", will that direct the student or cause extra confusion? This is probably a complicated authoring issue and I do hope that you will allow the question author to declare that they do not want to have all parts shown at once as they may have intentionally phased the parts according to their own plan.

Basically, you need to have the technical capability to draw a part that does not yet have all the possible parameters defined (i.e. those constructed from input to previous steps), and if you do not have that capability you cannot show any parts containing output constructed from input.

Finally, if we have blank spots in the latter steps that only get filled in as the student progresses, will those values appearing beyond the horizon (i.e. the next logical step) steal the student's focus? And how much this will differ based on the viewport size...
